leader.go

```diff
@@ -85,9 +86,22 @@ func (k *Kinsumer) becomeLeader() {
 
 // unbecomeLeader stops the leadership goroutine.
 func (k *Kinsumer) unbecomeLeader() {
+	// block until we know we're not unbecoming leader already
+	iteration := 0
+	for {
+		if !k.unbecomingLeader { // TODO: does looping indefinitely like this have any negative effects? should we sleep for 50ms in the loop?
```

Comment:
When you sleep in a thread, you give other threads the opportunity to do some work. So you're still blocking execution of this particular thread but your only checking your condition every x milestones rather than as fast as possible, which gives the cpu a little more time to do other tasks in between checks.